### ISSUE TITLE
False positive InvalidTemplateParam when use "static" as Template Param

### ISSUE THREAD

**Issue by darkdarin** [vimeo/psalm#11199]
Example:
https://psalm.dev/r/f8c1600e03

Result:
ERROR: [InvalidTemplateParam](https://psalm.dev/183) - 20:8 - Extended template param TModel expects type Model, type static given

Expects:
No errors.

Phpstan example: https://phpstan.org/r/b6e3f530-00ef-4bf5-9610-59bacad09f8d

**Comment by alies-dev** [comment]
Hitting the same root cause on Psalm 6.16.1, but the surface is broader than the trait `@use` form in the original report. The same false-positive `Extended template param ... expects type ..., type self/static given` reproduces with both `self` and `static`, and across all of `@implements`, `@extends`, `@template-implements`, and `@template-extends` (in addition to trait `@use`).

Synthetic repro (no errors expected; all eight class declarations describe a class that satisfies `T of Base`):

```php
<?php declare(strict_types=1);

abstract class Base {}

/** @template T of Base */
interface Holder {}

/** @template T of Base */
trait HolderTrait {}

/** @template T of Base */
abstract class ParentHolder extends Base {}

/** @implements Holder<self> */                  // error
final class ImplementsSelf extends Base implements Holder {}

/** @implements Holder<static> */                // error
final class ImplementsStatic extends Base implements Holder {}

/** @template-implements Holder<self> */         // error
final class TemplateImplementsSelf extends Base implements Holder {}

/** @extends ParentHolder<self> */               // error
final class ExtendsSelf extends ParentHolder {}

/** @extends ParentHolder<static> */             // error
final class ExtendsStatic extends ParentHolder {}

/** @template-extends ParentHolder<self> */      // error
final class TemplateExtendsSelf extends ParentHolder {}

final class TraitUseSelf extends Base {
    /** @use HolderTrait<self> */                // error
    use HolderTrait;
}

/** @implements Holder<ImplementsFqn> */         // no error (control)
final class ImplementsFqn extends Base implements Holder {}
```

Suspected source: `src/Psalm/Internal/Analyzer/ClassLikeAnalyzer.php`, `checkTemplateParams()`, around line 781 in 6.16.1. `$extended_type` is taken raw from `$storage->template_extended_params[...]` and compared via `UnionTypeComparator::isContainedBy(...)` against the parent's bound without first being passed through `TypeExpander::expandUnion(..., $self_class: $storage->name, $static_class_type: $storage->name, ...)`. Other call sites in the codebase (e.g. `Codebase\ClassLikes::1078`, `Codebase\Methods::750`) run that expansion before similar bound checks; this site was missing it.

Opened #11833 against `6.x` with the fix and 10 regression tests covering the matrix above plus negative cases (the bound check must still enforce when violated).
